packages/block-editor/src/hooks/child-layout.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalToggleGroupControl as ToggleGroupControl,
+	__experimentalToggleGroupControlOption as ToggleGroupControlOption,
+	__experimentalUnitControl as UnitControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import useSetting from '../components/use-setting';
+
+/**
+ * Inspector control panel containing the child layout related configuration
+ *
+ * @param {Object} props                        Block props.
+ * @param {Object} props.attributes             Block attributes.
+ * @param {Object} props.setAttributes          Function to set block attributes.
+ * @param {Object} props.__unstableParentLayout
+ *
+ * @return {WPElement} child layout edit element.
+ */
+export function ChildLayoutEdit( {
+	attributes,
+	setAttributes,
+	__unstableParentLayout: parentLayout,
+} ) {
+	const { style = {} } = attributes;
+	const { layout: childLayout = {} } = style;
+	const { selfStretch, flexSize } = childLayout;
+
+	return (
+		<>
+			<ToggleGroupControl
+				style={ { marginBottom: '8px' } }
```

Comment:
Override for an [overly-specific ToolsPanelItem style](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/tools-panel/styles.ts#L119).

Reply:
Thanks for advising on a preferred approach here @mirka 👍 

If I understand the suggestion correctly, the proposal is to update the BaseControl to be connected to the context system, through which we could then get the ToolsPanel to flag that all inner base controls shouldn't have a bottom margin.

Even if we were to implement that approach, this ad hoc compound component would need to reintroduce the margin only when the fixed width unit control is being rendered, i.e. still use the VStack approach recommended.

I don't think we'll benefit much from implementing the context system approach between now and when all the components have no margin by default My vote would be to wait for that and then remove the ToolsPanel styles.

> Override for an [overly-specific ToolsPanelItem style](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/tools-panel/styles.ts#L119).

FWIW the styles here were only as specific as required to actually overcome the BaseControl's margins at the time they were added. I still see them as better than doubling up on the spacing when combining margins with the panel grid's gap. Which is harder to avoid when some controls/components have margins or don't by default. The easiest option appeared to be getting back to a clean slate.